tools/publisher/src/subcommand/yank_release.rs

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::cargo;
+use anyhow::{bail, Context, Result};
+use clap::Parser;
+use dialoguer::Confirm;
+use regex::Regex;
+use smithy_rs_tool_common::shell::ShellOperation;
+use smithy_rs_tool_common::versions_manifest::{Release, VersionsManifest};
+use std::path::{Path, PathBuf};
+use std::str::FromStr;
+use std::sync::Arc;
+use tokio::sync::Semaphore;
+use tracing::info;
+
+const MAX_CONCURRENCY: usize = 5;
+
+#[derive(Parser, Debug)]
+pub struct YankReleaseArgs {
+    /// The aws-sdk-rust release tag to yank
+    #[clap(long, required_unless_present = "versions-toml")]
+    release_tag: Option<String>,
+    /// Path to a `versions.toml` file with a `[release]` section to yank.
+    /// The `--release-tag` option is preferred to this, but this is provided as a fail safe.
+    #[clap(long, required_unless_present = "release-tag")]
+    versions_toml: Option<PathBuf>,
+}
+
+pub async fn subcommand_yank_release(
+    YankReleaseArgs {
+        release_tag,
+        versions_toml,
+    }: &YankReleaseArgs,
+) -> Result<()> {
+    // Make sure cargo exists
+    cargo::confirm_installed_on_path()?;
+
+    // Retrieve information about the release to yank
+    let release = match (release_tag, versions_toml) {
+        (Some(release_tag), None) => acquire_release_from_tag(release_tag).await,
+        (None, Some(versions_toml)) => acquire_release_from_file(versions_toml),
+        _ => bail!("Only one of `--release-tag` or `--versions-toml` should be provided"),
+    }
+    .context("failed to retrieve information about the release to yank")?;
+
+    // Don't proceed unless the user confirms the plan
+    confirm_plan(&release)?;
+
+    // Use a semaphore to only allow a few concurrent yanks
+    let semaphore = Arc::new(Semaphore::new(MAX_CONCURRENCY));
+    info!(
+        "Will yank {} crates in parallel where possible.",
+        MAX_CONCURRENCY
+    );
+
+    let mut tasks = Vec::new();
+    for (crate_name, crate_version) in release.crates {
+        let permit = semaphore.clone().acquire_owned().await.unwrap();
+        tasks.push(tokio::spawn(async move {
+            info!("Yanking `{}-{}`...", crate_name, crate_version);
+            let result = cargo::Yank::new(&crate_name, &crate_version).spawn().await;
+            drop(permit);
+            info!("Successfully yanked `{}-{}`", crate_name, crate_version);
```

Comment:
this log line isn't correct I think? if the yank fails, we'll still see success here? probably need to use `result`

Reply:
Good catch!